src/main/scala/org/broadinstitute/dsde/workbench/avram/db/CollectionComponent.scala

```diff
@@ -0,0 +1,65 @@
+package org.broadinstitute.dsde.workbench.avram.db
+
+import java.sql.Timestamp
+import java.time.Instant
+
+import org.broadinstitute.dsde.workbench.avram.api.Collection
+import AvramPostgresProfile.api._
+
+import scala.concurrent.ExecutionContext
+
+case class CollectionRecord(id: Long,
+                            name: String,
+                            createdBy: String,
+                            samResource: String,
+                            createdTimestamp: Timestamp,
+                            updatedBy: String,
+                            updatedTimestamp: Timestamp)
+
+trait CollectionComponent extends AvramComponent {
+
+  class CollectionTable(tag: Tag) extends Table[CollectionRecord](tag, "collection") {
+    def id =                     column[Long]            ("id",                  O.PrimaryKey, O.AutoInc)
+    def name =                   column[String]          ("name",                O.Unique, O.Length(1000))
+    def samResource =            column[String]          ("sam_resource",        O.Length(1000))
+    def createdBy =              column[String]          ("created_by",          O.Length(1000))
+    def createdTimestamp =       column[Timestamp]       ("created_timestamp",   O.SqlType("TIMESTAMP(6)"))
+    def updatedBy =              column[String]          ("updated_by",          O.Length(1000))
+    def updatedTimestamp =       column[Timestamp]       ("updated_timestamp",   O.SqlType("TIMESTAMP(6)"))
+
+    def * = (id, name, samResource, createdBy, createdTimestamp, updatedBy, updatedTimestamp) <> (CollectionRecord.tupled, CollectionRecord.unapply)
+  }
+
+  object collectionQuery extends TableQuery(new CollectionTable(_)) {
+
+    def save(name: String, samResource: String, createdBy: String): DBIO[Int] = {
+      val now = Timestamp.from(Instant.now)
+      //currently the updatedBy and updatedTimestamp are the same as created at save time
+      collectionQuery += CollectionRecord(0, name, createdBy, samResource, now, createdBy, now)
```

Comment:
Not fond of the magic `0` and having to know that it's ignored. (I wonder if it's still ignored if we don't specify `O.AutoInc` on the column?) http://slick.lightbend.com/doc/3.2.3/queries.html looks to propose making the ID column an `Option` which makes this a little better (`None` instead of `0`). It also makes the custom * projection useful because you can use `id.?` so only the case class says `Option`, not the `Table`.

Reply:
I realize that I misunderstood a little about the * projection. It is currently being used to convert between the tuple of values and the `CollectionRecord` case class. The point though is that it doesn't have to be a 1-1 mapping. We are allowed to do other things here if it helps bridge the object-relational gap.

Reply:
I think using a 0 for now is fine, we can talk about it as a future improvement to the code, but I don't want to explore it for this PR.